File: rust-api/src/schemas.rs

use serde::{Serialize, Deserialize};
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct User {
    pub id: String,
    pub username: String,
    pub password: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Skill {
    pub name: String,
    pub level: u32,
    pub description: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Experience {
    pub company: String,
    pub position: String,
    pub description: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Education {
    pub school: String,
    pub degree: String,
    pub description: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Project {
    pub name: String,
    pub description: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Certificate {
    pub name: String,
    pub description: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Profile {
    pub id: Uuid,
    pub name: String,
    pub description: String,
    pub age: u32,
    pub email: String,
    pub company: String,
    pub skills: Vec<Skill>,
    pub experience: Vec<Experience>,
    pub education: Vec<Education>,
    pub projects: Vec<Project>,
    pub certificates: Vec<Certificate>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ProfileNoId {
    pub name: String,
    pub description: String,
    pub age: u32,
    pub email: String,
    pub company: String,
    pub skills: Vec<Skill>,
    pub experience: Vec<Experience>,
    pub education: Vec<Education>,
    pub projects: Vec<Project>,
    pub certificates: Vec<Certificate>,
}

File: python-api/schemas.py
from typing import List
from uuid import UUID
from pydantic import BaseModel
from dataclasses import dataclass


@dataclass
class UserNoId:
    username: str
    password: str


@dataclass
class User:
    id: UUID
    username: str
    password: str


@dataclass
class Skill:
    name: str
    level: int
    description: str


@dataclass
class Experience:
    company: str
    position: str
    description: str


@dataclass
class Education:
    school: str
    degree: str
    description: str


@dataclass
class Project:
    name: str
    description: str


@dataclass
class Certificate:
    name: str
    description: str


@dataclass
class Profile:
    id: UUID
    name: str
    description: str
    age: int
    email: str
    company: str
    skills: List[Skill]
    experience: List[Experience]
    education: List[Education]
    projects: List[Project]
    certificates: List[Certificate]


@dataclass
class ProfileNoId:
    name: str
    description: str
    age: int
    email: str
    company: str
    skills: List[Skill]
    experience: List[Experience]
    education: List[Education]
    projects: List[Project]
    certificates: List[Certificate]
